src/Featurizers/Components/GlobalMinimumFrequencyEstimator.h

```diff
@@ -0,0 +1,222 @@
+// ----------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License
+// ----------------------------------------------------------------------
+#pragma once
+
+#include <limits>
+
+#include "TrainingOnlyEstimatorImpl.h"
+#include "FrequencyEstimator.h"
+#include "GrainFeaturizerImpl.h"
+
+namespace Microsoft {
+namespace Featurizer {
+namespace Featurizers {
+namespace Components {
+
+// TODO: This should be removed in favor of the StatisticalMetricsEstimator
+
+static constexpr char const * const         GlobalMinimumFrequencyEstimatorName("GlobalMinimumFrequencyEstimator");
+static constexpr char const * const         GrainedFrequencyEstimatorName("GrainedFrequencyEstimator");
+
+using GrainedFrequencyEstimator = GrainEstimatorImpl<std::string, FrequencyEstimator<>>;

```

Comment:
>std::string [](start = 53, length = 11)

To fit with existing conventions, grain should be represented as `std::vector<std::string>`

Reply:
Will resolve in offline discussion.

Reply:
Resolved. Change `std::unordered_map` to `std::map` to avoid compiler complaints about using `std::vector<std::string>>` as key type.